stored_requests/backends/db_provider/postgres_dbprovider_test.go

```diff
@@ -0,0 +1,62 @@
+package db_provider
+
+import (
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/prebid/prebid-server/config"
+)
+
+func TestConnStringPostgres(t *testing.T) {
+	driver := "postgres"
+	db := "TestDB"
+	host := "somehost.com"
+	port := 20
+	username := "someuser"
+	password := "somepassword"
+
+	cfg := config.DatabaseConnection{
+		Driver:   driver,
+		Database: db,
+		Host:     host,
+		Port:     port,
+		Username: username,
+		Password: password,
+	}
+
+	provider := PostgresDbProvider{
+		cfg: cfg,
+	}
+	dataSourceName := provider.ConnString()
+	paramList := strings.Split(dataSourceName, " ")
+	params := make(map[string]string, len(paramList))
+	for _, param := range paramList {
+		keyVals := strings.Split(param, "=")
+		if len(keyVals) != 2 {
+			t.Fatalf(`param "%s" must only have one equals sign`, param)
+		}
+		if _, ok := params[keyVals[0]]; ok {
+			t.Fatalf("found duplicate param at key %s", keyVals[0])
+		}
+		params[keyVals[0]] = keyVals[1]
+	}
+
+	assertHasValue(t, params, "dbname", db)
+	assertHasValue(t, params, "host", host)
+	assertHasValue(t, params, "port", strconv.Itoa(port))
+	assertHasValue(t, params, "user", username)
+	assertHasValue(t, params, "password", password)
+	assertHasValue(t, params, "sslmode", "disable")
+}
+
+func assertHasValue(t *testing.T, m map[string]string, key string, val string) {
+	t.Helper()
+	realVal, ok := m[key]
+	if !ok {
+		t.Errorf("Map missing required key: %s", key)
+	}
+	if val != realVal {
+		t.Errorf("Unexpected value at key %s. Expected %s, Got %s", key, val, realVal)
+	}
+}
```

Comment:
The reason why I put these tests into separate files is to completely separate the logic because different implementations of `DbProvider` can have different unique parameters that produce connection strings in a variety of formats. Keeping a unified test for all of them sounds reasonable but can turn into a mesh in future. For example, MongoDB can have multiple hosts with ports in a connection string for a replica set and a sharded cluster.

@bsardo, please confirm if you think I should make it more unified anyway.

Reply:
@victorshevtsov oh maybe we have a misunderstanding; sorry about that. I wasn't questioning your test organization nor suggesting that we merge the connection string tests for postgres and mysql. I was just suggesting two things:
1. I think a lot of `db_provider/postgres_dbprovider_test.go#TestConnStringPostgres` can be deleted so that the test structure is similar to that of `db_provider/mysql_dbprovider_test.go#TestConnStringMySql`. That means the logic in `TestConnStringPostgres` that iterates over the `paramList` checking for equal sign occurrences and duplicate keys can be deleted and the calls to `assertHasValue` can be replaced with simpler assertion logic similar to that in `TestConnStringMySql`.
2. I think both `db_provider/postgres_dbprovider_test.go#TestConnStringPostgres` and `db_provider/mysql_dbprovider_test.go#TestConnStringMySql` should maybe be converted to a table-based format, each with cases that set just one of the database connection fields (`db`, `host`, `port`, `username`, `password`) in isolation, with a final test case that has them all set as each test currently does.

Does that make sense?

Reply:
Ah, I see now. Sorry for misunderstanding.
Yes, that makes sense.

Reply:
Done.